### ISSUE TITLE
Character count message should be placed correctly in markup without Javascript

### ISSUE THREAD

**Issue by hannalaakso** [alphagov/govuk-frontend#1602]
The character count Javascript [moves the count message](https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/components/character-count/character-count.js#L31) ("You have x characters remaining") in the DOM to be inside `govuk-form-group`. This was probably useful when the message markup was created dynamically. However this functionality is now redundant. The JS should be removed and the count message markup placed inside `govuk-form-group` in the component markup. This is so that we don't unnecessarily rely on Javascript for the correct order of DOM elements.

It’s important that the count message markup is inside `govuk-form-group` so that it gets wrapped within the error block (see screen grab below).

![65713384-fc0af600-e090-11e9-90d3-33230813d538](https://user-images.githubusercontent.com/5007934/66323584-ebbe0b00-e91b-11e9-9bbd-c367ed85530b.png)

We put off making this change this in https://github.com/alphagov/govuk-frontend/pull/1594 as it's a breaking change. 

## Done when

- [ ] TBC
- [ ] "Details of breaking change" section below completed to help with writing release notes

## Details of breaking change

- which users are affected: TBC
- the change that’s been made: TBC
- changes users will have to make: TBC
- impact of users not making those changes: TBC

**Comment by 36degrees** [comment]
I think this would require us to add something to the textarea API that allows us to insert this inside the form-group.

**Comment by hannalaakso** [comment]
Yes, we’d need to add it to the textarea API and tweak the textarea macro template so that it can accept a hint which would render below `<textarea>` but inside `.govuk-form-group`. But that is actually quite a strange requirement since the character count component is the only thing that needs to do this and most of the time inserting a hint below the textarea for any other purposes is not something we'd recommend teams do.

In conclusion, I don't think we should do this - it's a bit of an irregularity (ideally the JS wouldn’t need to move the markup around) but it's not causing any problems and if anything, adding it into the textarea API could actually lead to it being misused in other contexts. 

Tempted to just close this issue but leave it open for a bit longer for any other comments.

**Comment by lfdebrux** [comment]
If we made this change, would that mean that the hint would appear below the text area for users who have turned off JavaScript? That would seem like surprising/wrong behaviour to me.

**Comment by 36degrees** [comment]
> If we made this change, would that mean that the hint would appear below the text area for users who have turned off JavaScript? That would seem like surprising/wrong behaviour to me.

No. The hint and error message are standard, there's nothing special going on with them. However, at the minute the character count 'info message' ('You can enter up to X characters', which is separate to the hint) is outside of the form group in the component HTML, and the JS moves it inside the form group. 

That means that when JS is disabled and the input is in the error state, the info message appears outside of it:

<img width="1004" alt="Screenshot 2021-09-06 at 16 44 45" src="https://user-images.githubusercontent.com/121939/132240972-63dd50c4-ef47-4037-a370-5a85163d833b.png">

It also means we have to 'move' the margin that's normally applied to the form group for all (?) other form inputs from the form group to the character count wrapper:

https://github.com/alphagov/govuk-frontend/blob/133ea4f097ffb5b280357f7f8d75a8fff87a4d64/src/govuk/components/character-count/_index.scss#L7-L14

(Otherwise there would be a 30px margin between the form group and the info message, and only a small margin between the info message and the next component)

**Comment by colinrotherham** [comment]
This is now ready to review as part of https://github.com/alphagov/govuk-frontend/pull/4566
